view/src/components/todo.js

```diff
@@ -2,25 +2,422 @@ import React, { Component } from "react";
+          >
+            <AppBar className={classes.appBar}>
+              <Toolbar>
+                <IconButton
+                  edge="start"
+                  color="inherit"
+                  onClick={handleClose}
+                  aria-label="close"
+                >
+                  <CloseIcon />
+                </IconButton>
+                <Typography variant="h6" className={classes.title}>
+                  {this.state.buttonType === "Edit"
+                    ? "Edit Todo"
+                    : "Create a new Todo"}
+                </Typography>
+                <Button
+                  autoFocus
+                  color="inherit"
+                  onClick={handleSubmit}
+                  className={classes.submitButton}
+                >
+                  {this.state.buttonType === "Edit" ? "Save" : "Submit"}
+                </Button>
+              </Toolbar>
+            </AppBar>
+
+            <form className={classes.form} noValidate>
+              <Grid container spacing={2}>
+                <Grid item xs={12}>
+                  <TextField
+                    variant="outlined"
+                    required
+                    fullWidth
+                    id="todoTitle"
+                    label="Todo Title"
+                    name="title"
+                    autoComplete="todoTitle"
+                    helperText={errors.title}
+                    value={this.state.title}
+                    error={errors.title ? true : false}
+                    onChange={this.handleChange}
+                  />
+                </Grid>
+                <Grid item xs={12}>
+                  <TextField
+                    variant="outlined"
+                    required
+                    fullWidth
+                    id="todoDetails"
+                    label="Todo Details"
+                    name="body"
+                    autoComplete="todoDetails"
+                    multiline
+                    rows={25}
+                    rowsMax={25}
+                    helperText={errors.body}
+                    error={errors.body ? true : false}
+                    onChange={this.handleChange}
+                    value={this.state.body}
+                  />
+                </Grid>
+              </Grid>
+            </form>
+          </Dialog>
+
+          <Grid container spacing={2}>
+            {this.state.todos.map((todo) => (
+              <Grid item xs={12} sm={6}>
+                <Card className={classes.root} variant="outlined">
+                  <CardContent>
+                    <Typography variant="h5" component="h2">
+                      {todo.title}
+                    </Typography>
+                    <Typography className={classes.pos} color="textSecondary">
+                      {dayjs(todo.createdAt).fromNow()}
+                    </Typography>
+                    <Typography variant="body2" component="p">
+                      {`${todo.body.substring(0, 65)}`}
```

Comment:
Oddly specific. :)  With proportional fonts, a fixed number of characters won't help you much.  If your goal is to avoid printing more than one line of the todo, I recommend using the following CSS instead:

```
white-space: nowrap;
overflow: hidden;
text-overflow: ellipsis;
```

Reply:
I tried adding this to the typography container holding the todo.body but it still didn't prevent overflow. Any other thoughts?